test/core/repository/repository.spec.ts

```diff
@@ -0,0 +1,833 @@
+import { afterAll, beforeAll, describe, expect, test } from 'bun:test';
+import { randomUUID } from 'crypto';
+import knex from 'knex';
+import { PassThrough, Stream, Transform } from 'stream';
+
+import { Table } from '#/core/database/table';
+import { Repository } from '#/core/repository/repository';
+import { CoreError } from '#/error/coreError';
+import type { AdvancedSearch } from '#/types/data/advancedSearch';
+
+const options = {
+    databaseName: process.env.MSSQL_DATABASE ?? '',
+    host: process.env.MSSQL_HOST ?? '',
+    port: 1433,
+    user: process.env.MSSQL_USER ?? '',
+    password: process.env.MSSQL_PASSWORD ?? ''
+};
+
+const testTable = 'unit_test_repository';
+
+const knexInstance = knex({
+    client: 'mssql',
+    connection: {
+        database: options.databaseName,
+        host: options.host,
+        port: options.port,
+        user: options.user,
+        password: options.password,
+        options: { encrypt: true }
+    }
+});
+
+const table = new Table(
+    options.databaseName,
+    testTable,
+    ['id', 'name', 'number', 'date', 'boolean'],
```

Comment:
Pourquoi ici ce n'est pas ?
```suggestion
    ['id', 'name', 'age', 'birth', 'bool', 'n'],
```